pkg/v1/google/options.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package google
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+)
+
+// WithTransport is a functional option for overriding the default transport
+// on a remote image
+func WithTransport(t http.RoundTripper) ListerOption {
```

Comment:
Not a comment on this PR per se, but I think we're going to end up with a bunch of copies of these the way we're going. I'd love to figure out how to reuse them while making it clear which reusable options are usable in which scenarios. I can think of a way of abusing interfaces with private methods, but that's pretty gross. 😷

Reply:
Yeah I wasn't sure how to reuse these in a nice way 👎

I figure it's not a ton of code, so some duplication here isn't terrible, but I agree this is less than ideal :)